src/util/tryCatch.ts

```diff
@@ -11,7 +11,7 @@ function tryCatcher(): any {
   }
 }
 
-export function tryCatch(fn: Function): Function {
+export function tryCatch<T extends Function>(fn: T): T {
   tryCatchTarget = fn;
-  return tryCatcher;
-};
\ No newline at end of file
+  return <any>tryCatcher;
```

Comment:
I know. What I'm saying is that `typeof tryCatchMethod` should not be the same as `typeof predicate`. The return type of these functions is different.